opcua/server/address_space.py

```diff
@@ -491,43 +491,75 @@ def empty(self):
 
     def dump(self, path):
         """
-        dump address space as binary to file
+        dump address space as binary to file; note that server must be stopped for this method to work
         """
-        s = shelve.open(path, "n", protocol = pickle.HIGHEST_PROTOCOL)
+        # prepare nodes in address space for being serialized
+        for nodeid in self._nodes.keys():
+            node = self._nodes[nodeid]
+
+            # if the node has a reference to a method call, remove it so the object can be serialized
+            if hasattr(node, "call"):
+                self._nodes[nodeid].call = None
+
+        with open(path, 'wb') as f:
+            pickle.dump(self._nodes, f, pickle.HIGHEST_PROTOCOL)
+
+    def load(self, path):
+        """
+        load address space from file, overwriting everything in current address space
+        """
+        with open(path, 'rb') as f:
+            self._nodes = pickle.load(f)
+
+    def make_cache(self, path):
+        """
+        make a shelve for storing nodes from the standard address space; nodes are moved to a cache
+        by the LazyLoadingDict class when they are accessed
+        """
+        s = shelve.open(path, "n", protocol=pickle.HIGHEST_PROTOCOL)
         for nodeid in self._nodes.keys():
             s[nodeid.to_string()] = self._nodes[nodeid]
         s.close()
 
-    def load(self, path):
+    def load_cache(self, path):
```

Comment:
make_cache only creates the python shelve with the standard address space. Load cache instantiates the lazy loader which uses the shelve make_cache created.